pkgs/os-specific/linux/kernel-headers/mips-darwin.patch

```diff
@@ -0,0 +1,50 @@
+--- a/arch/mips/boot/tools/relocs.h
++++ b/arch/mips/boot/tools/relocs.h
+@@ -11,9 +11,16 @@
+ #include <errno.h>
+ #include <unistd.h>
+ #include <elf.h>
+-#include <byteswap.h>
+ #define USE_BSD
+-#include <endian.h>
++#if defined(__APPLE__)
++# include <libkern/OSByteOrder.h>
++# define bswap_16 OSSwapInt16
```

Comment:
Could we avoid having to patch the kernel by creating a trivial derivation that puts something like this in `$out/include/byteswap.h` and then add that trivial derivation as an input if `stdenv.buildPlatform.isDarwin`?

```
# include <libkern/OSByteOrder.h>
# define bswap_16 OSSwapInt16
# define bswap_32 OSSwapInt32
# define bswap_64 OSSwapInt64
```

It looks like `OSSwapIntXX` are just macros, with no linkage, although I don't have a darwin machine so I can't check this.

Reply:
Ah, great idea! This seems like a much cleaner solution, and doesn't require any patching.